### PR TITLE
Update NBGL to support Nanos

### PR DESCRIPTION
**Pull request by nroggeman-ledger**
## Description

The goal of this PR is to support NBGL on NanoS+ and NanoX, for OS and Application development.
Applications don't need any change and can still be compiled against this SDK and run on a Bolos using it 

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [*] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

## Breaking change

All apps need to be rebuilt because memory/flash maaping has changed (Cx Trampoline for example)

### CODE REVIEW

**Comment by xchapron-ledger** [comment]
More generally, it would be good to see what it would looks like on let's say app-boilerplate. To confirm that the use_case API is relevant and close enough to the Stax one. I'm wondering if we shouldn't make the API a bit more future proof, e.g. common so that if there are UX changes they are handled in the SDK side (expect for long wording maybe)?

File: lib_nbgl/include/nbgl_use_case.h [hunk 1]
```diff
@@ -131,13 +142,30 @@ void nbgl_useCasePlugInHome(const char                *plugInName,
...
                             nbgl_callback_t            quitCallback);
+#else   // TARGET_STAX
+void nbgl_useCaseHome(const char                *appName,
+                      const nbgl_icon_details_t *appIcon,
+                      const char                *appVersion,
+                      const char                *tagline,
+                      nbgl_callback_t            aboutCallback,
+                      nbgl_callback_t            quitCallback);
+#endif  // TARGET_STAX
+#ifdef TARGET_STAX
```
[the remark this answers is not in the document]
Comment (nroggeman-ledger) [reply lib_nbgl/include/nbgl_use_case.h:153]: Agreed

File: lib_nbgl/include/nbgl_use_case.h [hunk 2]
```diff
@@ -179,8 +207,28 @@ void nbgl_useCaseAddressConfirmation(const char *address, nbgl_choiceCallback_t
 void nbgl_useCaseAddressConfirmationExt(const char                      *address,
                                         nbgl_choiceCallback_t            callback,
                                         const nbgl_layoutTagValueList_t *tagValueList);
+#else   // HAVE_SE_TOUCH
+void nbgl_useCaseHome(const char                *appName,
+                      const nbgl_icon_details_t *appIcon,
+                      const char                *appVersion,
```
[the remark this answers is not in the document]
Comment (nroggeman-ledger) [reply lib_nbgl/include/nbgl_use_case.h:196]: Duly noted for a coming PR :)

File: lib_nbgl/doc/mainpage_nanos.dox [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#ifndef HAVE_SE_TOUCH
```
Comment (iartemov-ledger) [inline lib_nbgl/doc/mainpage_nanos.dox:1]: The file is not referenced from `doc/mainpage.dox`, is not it? 

Is it supposed to be a conditional inclusion protected with `#ifdef HAVE_SE_TOUCH`? (then we need to do so and remove `(Stax only)` from `doc/mainpage.dox`? And also to add `HAVE_SE_TOUCH` to PREDEFINED section of `doc/Doxyfile` to build the documentation for Stax by default? 

Or maybe it's better to have 2 `Screen drawing API` sections  - one for Stax and one for Nano and to build all documentation unconditionally?

File: lib_nbgl/doc/nbgl_step.dox [hunk 4]
```diff
@@ -0,0 +1,130 @@
+#ifdef NBGL_STEP
```
Comment (iartemov-ledger) [inline lib_nbgl/doc/nbgl_step.dox:1]: Should it be called `nbgl_step_nanos.dox` to highlight the fact that it is for the nanos ONLY? 
The same question for `nbgl_flow.dox`.

**Review by iartemov-ledger — COMMENTED** [review]
Beginning of documentation changes review.

**Review by abonnaudet-ledger — APPROVED** [review]

**Pull request merged**